Db.cs

```diff
@@ -29,31 +29,59 @@ public Db(IConfiguration configuration)
             Items = Client.GetContainer("replicache-sample-todo", "items");
         }
 
-        public async Task<UInt64> GetMutationID(String accountID, String clientID)
+        public async Task RegisterStoredProcedures()
         {
-            var def = new QueryDefinition("SELECT * FROM c WHERE c.accountID = @accountID AND c.id = @id")
-                .WithParameter("@accountID", accountID)
-                .WithParameter("@id", GetReplicacheStateID(clientID));
-            await foreach (ClientState state in Items.GetItemQueryIterator<ClientState>(def))
+            string sharedPath = "js/shared.js";
+            string utilSource = $"\n// {sharedPath}\n{File.ReadAllText(sharedPath)}";
+            string[] ids = { "spProcessMutation", "spGetMutationID" };
+            foreach (string id in ids)
             {
-                return state.LastMutationID;
+                var props = new StoredProcedureProperties
+                {
+                    Id = id,
+                    Body = File.ReadAllText($"js/{id}.js") + "\n" + utilSource
+                };
+                try
+                {
+                    await Items.Scripts.ReplaceStoredProcedureAsync(props);
+                }
+                catch (CosmosException ex)
+                {
+                    if (ex.Status == 404)
+                    {
+                        await Items.Scripts.CreateStoredProcedureAsync(props);
+                    }
+                    else
+                    {
+                        throw;
+                    }
+                }
+
             }
-            return 0;
         }
 
-        public async Task SetMutationID(String accountID, String clientID, UInt64 mutationID)
+        public async Task<UInt64> GetMutationID(String accountID, String clientID)
```

Comment:
This is not needed anymore, nor is `spGetMutationID`.

Reply:
This is still needed by the client view